### PR TITLE
Fix custom nested query class check

### PR DESCRIPTION
**Pull request by michalkleiner**
Use params in the subclass check in the right order.

### CODE REVIEW

**Comment by flamerohr** [comment]
Thanks @michalkleiner!

do you want to shift this to the 1 branch?
going to master will mean this won't be fixed until version 2

**Comment by michalkleiner** [comment]
Against 1.0 the change is a bit bigger as it backports some code added further along the way, but it should be fine.

File: src/Scaffolding/Scaffolders/DataObjectScaffolder.php [hunk 1]
```diff
@@ -495,9 +495,23 @@ public function applyConfig(array $config)
...
+                            'Tried to specify %s as a custom query scaffolder for %s on %s, but it is not a subclass of %s.',
+                            $settings,
+                            $relationName,
+                            $this->dataObjectClass,
+                            QueryScaffolder::class
+                        ));
+                    }
+                } else {
+                    $this->nestedQuery($relationName)
+                        ->applyConfig((array)$settings);
```
Comment (flamerohr) [inline src/Scaffolding/Scaffolders/DataObjectScaffolder.php]: I'm a little adverse to add that thrown Exception, as it changes expectations for a patch fix... might be a good idea to remove that for the 1.0 branch.

Can ignore my slack message about this line being skipped :)

**Comment by unclecheese** [comment]
We'll need test coverage for this. `nestedQuery($scaffolderInstance)` isn't tested anywhere from what I can see, so this would be a good time to pay off that technical debt. :)

Likewise, with this PR, the `applyConfig` test should be updated to include a FQCN scaffolder string. https://github.com/silverstripe/silverstripe-graphql/blob/2/tests/Scaffolding/Scaffolders/DataObjectScaffolderTest.php#L211

**Comment by michalkleiner** [comment]
@unclecheese what if we change the PR to target the version where the bug I was fixing has been introduced? Would the unit test be there already? And then when that gets merged to version 1, all will be nice and fluffy already.

**Comment by maxime-rainville** [comment]
We're focusing on the 4 branch right now. We're not looking at doing another 1.x release.

**Pull request closed without merging**